q2_minimap2/_filtering_utils.py

```diff
@@ -76,61 +66,55 @@ def get_alignment_length(cigar):
 
 
 # Function to process a SAM file, filter based on mappings and identity percentage
-def process_sam_file(input_sam_file, keep_mapped, min_per_identity):
-    # Creates a temporary file to write filtered alignments to
-    with tempfile.NamedTemporaryFile(mode="w", delete=False) as tmp_file:
-        temp_file_path = tmp_file.name
+def process_sam_file(input_sam_file, keep, min_per_identity):
+    # Creates a temporary file and opens the input SAM file for reading simultaneously
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as tmp_file, open(
+        input_sam_file, "r"
+    ) as infile:
 
-    # Opens the input SAM file and the temporary file for writing
-    with open(input_sam_file, "r") as infile, open(temp_file_path, "w") as outfile:
         for line in infile:
             # Writes header lines directly to the output file
             if line.startswith("@"):
-                outfile.write(line)
+                tmp_file.write(line)
                 continue
 
-            # Extract information from cigar
+            # Extract information from the line
             parts = line.split("\t")
             flag = int(parts[1])
             cigar = parts[5]
 
             # Calculates identity percentage for alignments with a valid CIGAR string
-            if min_per_identity is not None and cigar != "*":
+            if min_per_identity and cigar != "*":
                 total_length = get_alignment_length(cigar)
                 identity_percentage = calculate_identity(line, total_length)
             else:
-                # Defaults identity percentage to 1 (100%) if no CIGAR string or no
-                # min_per_identity is specified
+                # Defaults identity percentage to 100% if no CIGAR string or no
+                # min_per_identity specified
                 identity_percentage = 1
 
             # Logic for including or excluding reads based on mappings and
             # identity percentage
-            if keep_mapped:
-                # Keep reads if they are mapped
+            if keep == "mapped":
                 if not (flag & 0x4) and not (flag & 0x100):
-                    # If the minimum identity percentage is provided then accept the
-                    # mapped read if its identity percentage is equal or exceeds
-                    # the minimum identity percentage
-                    if (min_per_identity is None) or (
-                        identity_percentage >= min_per_identity
+                    if (
+                        min_per_identity is None
+                        or identity_percentage >= min_per_identity
                     ):
```

Comment:
```suggestion
                       if not min_per_identity or identity_percentage >= min_per_identity:
```
This fits on one line. I think it is easier to read